tests/conftest.py

```diff
@@ -1513,3 +1514,11 @@ def guard_testing_venv():
             venv_was_altered = True
             error_message += f"\t* {pkg}: initial version={version_before_tests} --> after tests={version_after_tests}\n"
     assert not venv_was_altered, error_message
+
+
+@pytest.fixture(scope="function", autouse=True)
```

Comment:
we have regretted every single autouse fixture we every introduced, I don't know what could go wrong here, but .....